live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/config/messaging/LiveIngesterMessagingConfig.java

```diff
@@ -62,20 +63,24 @@ public Application application()
     }
 
     @Bean
-    public DiscoveryApiClient discoveryApiClient(IntegrationProperties integrationProperties, AuthService authService, ObjectMapper objectMapper)
+    public RepositoryInformation repositoryInformation(AuthService authService, ObjectMapper objectMapper, IntegrationProperties integrationProperties)
     {
-        return new DiscoveryApiClient(integrationProperties.alfresco().repository().discoveryEndpoint(), authService, objectMapper);
+        if (StringUtils.isNotBlank(integrationProperties.alfresco().repository().versionOverride()))
+        {
+            return new PropertyBasedRepositoryInformation(integrationProperties);
+        }
+        else if (StringUtils.isNotBlank(integrationProperties.alfresco().repository().discoveryEndpoint()))
+        {
+            return new DiscoveryApiRepositoryInformation(integrationProperties.alfresco().repository().discoveryEndpoint(), authService, objectMapper, HttpClient.newHttpClient());
+        }
+        throw new IllegalStateException("Either property alfresco.repository.discovery-endpoint or alfresco.repository.version-override must be set in the Live Ingester configuration.");
     }
 
     @Bean
-    public ApplicationInfoProvider applicationInfoProvider(DiscoveryApiClient discoveryApiClient, IntegrationProperties integrationProperties)
+    @DependsOn("repositoryInformation")
```

Comment:
I find it odd that this `DependsOn` was necessary even though `RepositoryInformation` is one of the required arguments (read: beans) of the method. Could this be an intermediate change that is not required anymore?

Reply:
Thanks for catching that - probably a leftover from some try-and-test - removing that annotation.